go/src/koding/kites/kloud/metadata/cloudinit.go

```diff
@@ -145,6 +154,34 @@ func mergeList(in, out []interface{}, path []string) ([]interface{}, error) {
 					}
 				}
 			}
+
+			// Check whether it's possible to merge-in
+			// the item.
+			for _, key := range mergeKeys {
+				val, ok := obj[key].(string)
+				if !ok || val == "" {
+					continue
+				}
+
+				for i := range out {
+					orig, ok := out[i].(map[string]interface{})
+					if !ok {
+						continue
+					}
+
+					if origVal, ok := orig[key].(string); !ok || origVal != val {
+						continue
+					}
+
+					if err := merge(shallowCopy(obj, key), orig); err != nil {
+						return nil, err
+					}
+
+					out[i] = orig
+
+					continue loop
```

Comment:
is there any way to rewrite this without "continue loop" making really hard to read & understand the flow?